gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -150,117 +201,300 @@ def test_unauthorized_user_cannot_list_tes_tasks(self):
+        """
+        Test Case: Verify that a Nextflow workflow can be executed successfully.
+        """
+        expected_task_outputs = {
+            "extract_metadata (1)": {
+                "filename": "dicom-metadata-img-1.dcm.csv",
+                "command": "python3 /utils/extract_metadata.py img-1.dcm",
+            },
+            "extract_metadata (2)": {
+                "filename": "dicom-metadata-img-2.dcm.csv",
+                "command": "python3 /utils/extract_metadata.py img-2.dcm",
+            },
+            "dicom_to_png (1)": {
+                "filename": "img-1.png",
+                "command": "python3 /utils/dicom_to_png.py img-1.dcm",
+            },
+            "dicom_to_png (2)": {
+                "filename": "img-2.png",
+                "command": "python3 /utils/dicom_to_png.py img-2.dcm",
+            },
+        }
+        workflow_dir = "gen3-integration-tests/test_data/gen3_workflow/"
+        workflow_log = self.gen3_workflow.run_nextflow_task(
+            workflow_dir=workflow_dir,
+            workflow_script="main.nf",
+            nextflow_config_file="nextflow.config",
+        )
+
+        completed_tasks = [
+            self._nextflow_parse_completed_line(line)
+            for line in workflow_log.split("\n")
+            if "Task completed > TaskHandler" in line
+        ]
+
+        for task in completed_tasks:
+
+            task_name = task["process_name"]
+            assert (
+                task_name in expected_task_outputs
+            ), f"Unexpected task name: {task_name}. Expected one of {list(expected_task_outputs.keys())}"
+
+            expected = expected_task_outputs[task_name]
+
+            assert (
+                task["status"] == "COMPLETED"
+            ), f"Task {task_name} failed with status: {task['status']}"
+            assert (
+                task["exit_code"] == "0"
+            ), f"Task {task_name} failed with exit code: {task['exit_code']}"
+            assert (
+                task["workDirProtocol"] == "s3"
+            ), f"Expected workDir to be an 's3' location, but got {task['workDirProtocol']}"
+
+            s3_file_list = self.gen3_workflow.list_bucket_objects_with_boto3(
+                folder_path=task["workDir"],
+                s3_storage_config=self.s3_storage_config,
+                user=self.valid_user,
+            )
+
+            assert (
+                s3_file_list
+            ), f"Expected to find files in the S3 bucket for task {task_name}, but got an empty list"
+
+            filenames_in_s3 = {file["Key"].split("/")[-1] for file in s3_file_list}
+
+            assert (
+                expected["filename"] in filenames_in_s3
+            ), f"Expected to find {expected['filename']} in the S3 bucket for task {task_name}, but got {s3_file_list}"
+
+            command_file_keys = [
+                file["Key"]
+                for file in s3_file_list
+                if file["Key"].endswith(".command.sh")
```

Comment:
```suggestion
                if file["Key"].endswith("/.command.sh")
```
maybe?

Reply:
I don’t think this is necessary, since with Nextflow we assume there won’t be a scenario where command files have different names. That said, I can add it since it doesn’t hurt. Let me know if you can think of any edge cases I might’ve missed.

Reply:
edge case: input or output file named `my.custom.command.sh`